contracts/ronin/validator/storage-fragments/TimingStorage.sol

```diff
@@ -16,13 +16,23 @@ abstract contract TimingStorage is ITimingInfo, GlobalConfigConsumer {
   uint256 internal _currentPeriodStartAtBlock;
 
   /// @dev Mapping from epoch index => period index
-  mapping(uint256 => uint256) internal _periodOf;
-
+  mapping(uint256 epoch => uint256 period) internal _periodOf;
+  /// @dev Mapping from period index => ending block
+  mapping(uint256 period => uint256 endedAtBlock) internal _periodEndBlock;
   /**
    * @dev This empty reserved space is put in place to allow future versions to add new
    * variables without shifting down storage in the inheritance chain.
    */
-  uint256[49] private ______gap;
+  uint256[48] private ______gap;
+
+  /**
+   * @inheritdoc ITimingInfo
+   */
+  function getPeriodEndBlock(uint256 period) external view returns (uint256 endedAtBlock) {
+    if (period > _lastUpdatedPeriod) revert ErrPeriodNotEndedYet(period);
+    endedAtBlock = _periodEndBlock[period];
+    if (endedAtBlock == 0) revert ErrPeriodEndingBlockNotExist(period);
```

Comment:
I prefer to store the period that begins tracking, then compare `period` with that starting tracking period. This == 0 may cause confusion if there is an actual non-exist period (happen when hang chain more than 1 day).